### PR TITLE
Eliane Weather App | Week 4

### PR DESCRIPTION
**Pull request by El1an3**
## Netlify link
https://elianes-weather-app.netlify.app/

## Collaborators
none, solo project

### CODE REVIEW

File: code/index.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  <body>
+    <header><p>Weather App</p></header>
+    <section class="app-container">
+      <!-- <div class="search-bar">
+        <label for="search">Search City:</label>
+        <input
+          id="search"
+          type="search"
+          placeholder="Type a city name here..."
+        />
```
Comment (sofia32057) [inline code/index.html:18]: Hey! Don't forget to remove commented out parts :)

File: code/index.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          <option value="New York">New York</option>
+          <option value="Lima">Lima</option>
+        </select>
+      </div>
+      <div id="weather-today"></div>
+      <div id="weather-description"></div>
+
+      <div id="fourday-forecast"></div>
+    </section>
+    <header>by Eliane for Technigo Bootcamp</header>
```
Comment (sofia32057) [inline code/index.html:37]: This should probably be at <footer>?

File: code/script.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+const getWeather = (city) => {
+  fetch(
+    `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&APPID=bac28b010cea73460ead078a7d8aa965`
+  )
+    .then((response) => response.json())
+    .then((data) => {
+      const sunriseTime = convertTo24Hour(data.sys.sunrise, data.timezone); //Unix timestamp (in seconds)
+      const sunsetTime = convertTo24Hour(data.sys.sunset, data.timezone); //Unix timestamp (in seconds)
+
+      weatherToday.innerHTML = `<p>${
```
Comment (sofia32057) [inline code/script.js:17]: Could this function be broken up into smaller parts?

File: code/script.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+};
+
+//get forecast data based on selected city
+const getForecast = (city) => {
+  fetch(
+    `https://api.openweathermap.org/data/2.5/forecast?q=${city}&units=metric&APPID=bac28b010cea73460ead078a7d8aa965`
+  )
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Network is not responding");
```
Comment (sofia32057) [inline code/script.js:93]: Nicely done with error handling! 💪

File: code/script.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    })
+    .then((data) => {
+      const dailyForecasts = {};
+
+      data.list.forEach((forecast) => {
+        const date = new Date(forecast.dt_txt);
+        //filtering for 12:00 forecast
+        if (date.getHours() === 12) {
+          const dayName = getDayName(date);
+          //If the day name doesn't already exist in the dailyForecasts object and the number of keys (representing the number of days) is less than 4, it stores the temperature for that day in the dailyForecasts object. This ensures that only forecasts for the next four days are stored.
```
Comment (sofia32057) [inline code/script.js:105]: Clever! And really good comment, thanks!

File: code/script.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+            };
+          }
+        }
+      });
+
+      //iterates over dailyForecast and generates HTML to display forecast
+      for (const dayName in dailyForecasts) {
+        fourdayForecast.innerHTML += `<div class="forecast-element"><div><p>${dayName}</p></div><div><p>${dailyForecasts[
+          dayName
+        ].temp.toFixed(1)}°C</p></div>`;
```
Comment (sofia32057) [inline code/script.js:121]: Didn't know about the .toFixed method. Very handy. Thank you!

File: code/script.js [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+      }
+    })
+    .catch((error) => {
+      console.error("There was a problem. Please try again later:", error);
+      fourdayForecast.innerHTML = `<p>There was a problem. Please try again later.`;
+    });
```
Comment (sofia32057) [inline code/script.js:127]: Awesome! 🥳🙌

File: code/script.js [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+  // Fetch weather and forecast data for the selected city
+  getWeather(selectedCity);
+  getForecast(selectedCity);
+});
+
+// Function to convert time to 24-hour format
+// added timezone to make sunsets/sunrise work
+const convertTo24Hour = (time, timezone) => {
+  //creates new Date object using the time parameter multiplied by 1000. Date expects the input to be in milliseconds, while Unix timestamps are in seconds.
+  const date = new Date((time + timezone) * 1000);
```
Comment (sofia32057) [inline code/script.js:148]: 🤩

File: code/script.js [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+  //extracting hours and minutes and converting to strings using UTCHHours tow make timezones work
+  //padStart adds leading zeros to ensure that hours/minutes always have two digits
+
+  const hours = date.getUTCHours().toString().padStart(2, "0");
+  const minutes = date.getUTCMinutes().toString().padStart(2, "0");
```
Comment (sofia32057) [inline code/script.js:153]: Clean!

File: code/style.css [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  font-weight: 400;
+}
+
+* {
+  box-sizing: border-box;
+  padding: 0;
+  margin: 0;
+}
+
+body.warm .app-container {
```
Comment (sofia32057) [inline code/style.css:15]: Clever with warm and cold classes!

**Review by sofia32057 — COMMENTED** [review]
Awesome Eliane! Your app looks and works GREAT. 

For next project, I would suggest trying to divide the logic into smaller functions, to be a little more flexible. But whatever works 🔥

File: code/script.js [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+        data.weather[0].description
+      } | ${data.main.temp.toFixed(
+        1
+      )}°C</p><p>sunrise ${sunriseTime}</p><p>sunset ${sunsetTime}</p>`;
+
+      //style based on temperature
+      if (data.main.temp <= 10) {
+        document.getElementsByTagName("body")[0].classList = ["cold"];
+      } else {
+        document.getElementsByTagName("body")[0].classList = ["warm"];
```
Comment (HIPPIEKICK) [inline code/script.js:27]: You can always reach the body by writing `document.body`

File: code/script.js (hunk 5 above)
Comment (HIPPIEKICK) [reply code/script.js:105]: Hmm, I don't think it's working as intended - it shows thu-sun but it should show fri-mon ⚠️

File: code/script.js (hunk 7 above)
Comment (HIPPIEKICK) [reply code/script.js:127]: Agree, good error handling 👍

File: code/script.js [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+  const hours = date.getUTCHours().toString().padStart(2, "0");
+  const minutes = date.getUTCMinutes().toString().padStart(2, "0");
+  return `${hours}:${minutes}`;
+};
+
+//function to get abbreviated day name
+const getDayName = (date) => {
+  const days = ["sun", "mon", "tue", "wed", "thu", "fri", "sat"];
+  return days[date.getDay()];
+};
```
Comment (HIPPIEKICK) [inline code/script.js:161]: ⭐

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Good job with the weather app Eliane! You followed most of the design and your code is quite structured too, I made some small comments of improvements.

One thing I need you to fix is the forecast, so that it displays the following four days (now it's displaying today + three days in the forecast) as per these requirements:
- The app should have: city name, current temperature, weather description, sunrise/sunset time, 4-day forecast
- The presentation of the data should be in the specified format

Feel free to reach out on SO if you need assistance.

File: code/index.html (hunk 2 above)
Comment (El1an3) [reply code/index.html:37]: It was a footer and then I was lazy and reused the header-styling ;-)

File: code/script.js (hunk 5 above)
Comment (El1an3) [reply code/script.js:105]: fixed

**Comment by HIPPIEKICK** [comment]
Now the forecast starts on Thursday instead of the coming day 👀

**Review by AntonellaMorittu — APPROVED** [review]

**Pull request closed without merging**